src/TestFramework/TestFramework/Attributes/TestMethod/TimeoutAttribute.cs

```diff
@@ -35,4 +37,21 @@ public TimeoutAttribute(TestTimeout timeout)
     /// Gets the timeout in milliseconds.
     /// </summary>
     public int Timeout { get; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the method should be cooperatively cancelled on timeout.
+    /// When set to <see langword="true"/>, the method should be designed to cooperatively cancel itself when the timeout is reached.
+    /// Otherwise, when set to <see langword="false"/>, the task method will be unobserved.
+    /// </summary>
+    public bool CooperativeCancellation
+    {
+        get => _isCooperativeCancellation;
+        set
+        {
+            IsCooperativeCancellationSet = true;
```

Comment:
Nullable boolean is not supported on attributes hence the trick with this other property to know this was set.

Reply:
```suggestion
            // Attributes don't allow nullable boolean, so we need this to know that the value was set explicitly.
            IsCooperativeCancellationSet = true;
```